docs/posts/bigquery-arrays/index.qmd

```diff
@@ -0,0 +1,247 @@
+
+We'll start with `from ibis.interactive import *` for maximum convenience.
+
+```{python}
+from ibis.interactive import *
+
+con = ibis.connect("bigquery://ibis-gbq")  # <1>
+con.set_database("bigquery-public-data.imdb")  # <2>
+```
+
+1. Connect to the **billing** project. Compute (but not storage) is billed to
+   this project.
+2. Set the database to the project and dataset that we will use for analysis.
+
+Let's look at the tables in this dataset:
+
+```{python}
+con.list_tables()
+```
+
+Let's pull out the `name_basics` table, which contains names and metadata about
+people listed on IMDB. We'll call this `ents` (short for `entities`), and remove some
+columns we won't need:
+
+```{python}
+ents = con.tables.name_basics.drop("birth_year", "death_year")
+ents
+```
+
+### Splitting strings into arrays
+
+We can see that `known_for_titles` looks sort of like an array, so let's call
+the
+[`split`](../../reference/expression-strings.qmd#ibis.expr.types.strings.StringValue.split)
+method on that column and replace the existing column:
+
+```{python}
+ents = ents.mutate(known_for_titles=_.known_for_titles.split(","))
+ents
+```
+
+Similarly for `primary_profession`, since people involved in show business often
+have more than one responsibility on a project:
+
+```{python}
+ents = ents.mutate(primary_profession=_.primary_profession.split(","))
+```
+
+### Array length
+
+Let's see how many titles each entity is known for, and then show the five
+people with the largest number of titles they're known for:
+
+This is computed using the
+[`length`](../../reference/expression-collections.qmd#ibis.expr.types.arrays.ArrayValue.length)
+API on array expressions:
+
+```{python}
+(
+    ents.select("primary_name", num_titles=_.known_for_titles.length())
+    .order_by(_.num_titles.desc())
+    .limit(5)
+)
+```
+
+It seems like the length of the `known_for_titles` might be capped at five!
+
+### Index
+
+We can see the position of `"actor"` in `primary_profession`s:
+
+```{python}
+ents.primary_profession.index("actor")
+```
+
+A return value of `-1` indicates that `"actor"` is not present in the value:
+
+Let's look for entities that are not primarily actors:
+
+We can do this using the
+[`index`](../../reference/expression-collections#ibis.expr.types.arrays.ArrayValue.index)
+method by checking whether the position of the string `"actor"` is greater than
+zero:
+
+```{python}
+actor_index = ents.primary_profession.index("actor")
+not_primarily_actors = actor_index > 0
+not_primarily_actors.mean()  # <1>
+```
+
+1. The average of a `bool` column gives the percentage of `True` values
+
+Who are they?
+
+```{python}
+ents[not_primarily_actors]
+```
+
+It's not 100% clear whether the order of elements in `primary_profession` matters here.
+
+### Containment
+
+We can get people who are **not** actors using `contains`:
+
+```{python}
+non_actors = ents[~_.primary_profession.contains("actor")]
```

Comment:
this line could use a quick comment/annotation/explanation somewhere IMO with the `~_.`, I think that could be confusing for a newer reader

Reply:
perhaps reference the selectors blog/documentation somewhere

Reply:
It's not a selector that I'm using here, this is a filter expression.